lib/os/mem_blocks.c

```diff
@@ -57,6 +57,45 @@ static int free_bits(sys_mem_blocks_t *mem_block, void *ptr, size_t bits)
 	return ret;
 }
 
+int sys_mem_blocks_alloc_continuous(sys_mem_blocks_t *mem_block, size_t count,
+				   void **out_block)
+{
+	int ret = 0;
+
+	if ((mem_block == NULL) || (out_block == NULL)) {
+		ret = -EINVAL;
+		goto out;
+	}
+
+	if (count == 0) {
+		/* Nothing to allocate */
+		*out_block = NULL;
+		goto out;
+	}
+
+	if (count > mem_block->num_blocks) {
+		/* Definitely not enough blocks to be allocated */
+		ret = -ENOMEM;
+		goto out;
+	}
+
+	void *ptr = alloc_bits(mem_block, count);
```

Comment:
I'm too lazy to check: is the existing alloc_bits() utility atomic?  If not, then you should probably add a lock here for symmetry with the other APIs being added.

Reply:
it is atomic, in fact it is using an atiomic procedure sys_bitarray_alloc